rpcserver.go

```diff
@@ -2853,31 +2910,64 @@ func ValidateAddress(w *Wallet, chainSvr *chain.Client, icmd btcjson.Cmd) (inter
 	result.Address = addr.EncodeAddress()
 	result.IsValid = true
 
-	ainfo, err := w.KeyStore.Address(addr)
-	if err == nil {
-		result.IsMine = true
-		result.Account = ""
-
-		if pka, ok := ainfo.(keystore.PubKeyAddress); ok {
-			result.IsCompressed = pka.Compressed()
-			result.PubKey = pka.ExportPubKey()
-
-		} else if sa, ok := ainfo.(keystore.ScriptAddress); ok {
-			result.IsScript = true
-			addresses := sa.Addresses()
-			addrStrings := make([]string, len(addresses))
-			for i, a := range addresses {
-				addrStrings[i] = a.EncodeAddress()
-			}
-			result.Addresses = addrStrings
-			result.Hex = hex.EncodeToString(sa.Script())
-
-			class := sa.ScriptClass()
-			// script type
-			result.Script = class.String()
-			if class == btcscript.MultiSigTy {
-				result.SigsRequired = int32(sa.RequiredSigs())
-			}
+	ainfo, err := w.Manager.Address(addr)
+	if managerErr, ok := err.(waddrmgr.ManagerError); ok {
+		if managerErr.ErrorCode == waddrmgr.ErrAddressNotFound {
```

Comment:
Drop this into the `if err != nil` below and add a `isAddressNotFoundError` method to match the other error code checks.